bin/varnishtest/tests/c00042.vtc

```diff
@@ -48,9 +48,79 @@ client c1 {
 	txreq -url /1
 	rxresp
 	expect resp.status == 200
+	expect resp.http.Authority == "127.0.0.1"
 	txreq -url /2
 	rxresp
 	expect resp.status == 200
+	expect resp.http.Authority == "127.0.0.1"
+} -run
+
+varnish v1 -vcl {
+	backend v2 { .host = "${v2_addr}"; .port = "${v2_port}"; }
+
+	backend s1 {
+		.via = v2;
+		.host = "${s1_addr}";
+		.port = "${s1_port}";
+		.authority = "authority.com";
+	}
+
+	sub vcl_recv {
+		set req.backend_hint = s1;
+	}
+}
+
+client c1 {
+	txreq -url /3
+	rxresp
+	expect resp.status == 200
+	expect resp.http.Authority == "authority.com"
+} -run
+
+varnish v1 -vcl {
+	backend v2 { .host = "${v2_addr}"; .port = "${v2_port}"; }
+
+	backend s1 {
+		.via = v2;
+		.host = "${s1_addr}";
+		.port = "${s1_port}";
+		.host_header = "host.com";
+	}
+
+	sub vcl_recv {
+		set req.backend_hint = s1;
+	}
+}
+
+client c1 {
+	txreq -url /4
+	rxresp
+	expect resp.status == 200
+	expect resp.http.Authority == "host.com"
+} -run
+
+# Setting .authority = "" disables sending the TLV.
+varnish v1 -vcl {
+	backend v2 { .host = "${v2_addr}"; .port = "${v2_port}"; }
+
+	backend s1 {
+		.via = v2;
+		.host = "${s1_addr}";
+		.port = "${s1_port}";
+		.authority = "";
+	}
+
+	sub vcl_recv {
+		set req.backend_hint = s1;
+	}
+}
+
+client c1 {
+	txreq -url /5
+	rxresp
+	expect resp.status == 200
+	# vmod_proxy returns the empty string if the TLV is absent.
+	expect resp.http.Authority == ""
 } -run
```

Comment:
Maybe rework the test case slightly so we can check the absence of authority with this construct:

```
expect resp.http.Authority == <unset>
```

Not sure whether `proxy.authority()` returns null to help make this case.